Changed tabs to spaces

czyzby opened the pull request:
* Changed tabs to spaces as per Android style guide.
* Updated project's `.editorconfig` to reflect the changes.
* Reformatted sources with `ktlint`.
* Added `ktlint` check to the build GitHub action.
* Added `.editorconfig` generation.
  * Its settings were added to the `AdvancedProjectData`, but keep in mind they are not hooked up to the UI yet, and changing the indent programmatically does not reformat the generated sources (it only affects the `.editorconfig` settings).

Frosty-J commented:
The style guide says [four spaces should be used](https://developer.android.com/kotlin/style-guide#indentation). This looks like two? (though I personally prefer two...)

czyzby commented:
Good catch, though I also find 2 readable enough and better for DSLs. For the record, I used 4 in the generated Java sources.

I'll leave it up to @tommyettinger whether we want to go for 2 or 4. We could consider using 2 throughout the `gdx-liftoff` and 4 in the generated projects.

czyzby commented:
I went ahead and changed the indent to 4 in the generated sources, also fixing some templates that I've missed along the way. I've also added `.editorconfig` to the generated projects, so that the IDE will automatically apply at least some of the appropriate formatting.

tommyettinger commented:
4 seems best for consistency, all around, but I'm totally fine if the liftoff sources use 2. I find both readable, and if the linter understands it, then using 4 spaces for generated sources should be enough to pass style guides.

tommyettinger commented on src/main/resources/generator/ios-moe/xcode/ios-moe.xcodeproj/project.pbxproj:
```diff
@@ -1,595 +1,595 @@
-		581773AE1E37A7EF004E28A9 /* UIKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 581773AD1E37A7EF004E28A9 /* UIKit.framework */; };
-		C995F7171EAA5D0A003E34E1 /* assets in Resources */ = {isa = PBXBuildFile; fileRef = C995F7161EAA5D0A003E34E1 /* assets */; };
-		E21F0FA41F8922D8003761DD /* Media.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = E21F0FA31F8922D8003761DD /* Media.xcassets */; };
+    093EA0331288EEC18D3C79EC /* main.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 1E152A74253836079FA1075F /* main.cpp */; };
```
Also, this nasty generated stuff looks like it might get regenerated using tabs... I'd have to check git blame.

czyzby replied:
The questions is whether tabs or spaces were used originally. We can revert those files to pre-formatting commits.

czyzby replied:
Reverted.

tommyettinger reviewed (approved):
This looks mostly good; I'm just not sure if a .pbxproj file (part of an XCode project) should be touched at all, since it looks like generated code.

czyzby commented on src/main/resources/generator/ios-moe/xcode/ios-moe-Test/Info.plist:
```diff
@@ -2,42 +2,42 @@
-		<string>UIInterfaceOrientationLandscapeLeft</string>
-		<string>UIInterfaceOrientationLandscapeRight</string>
-	</array>
+  <key>CFBundleDevelopmentRegion</key>
```
I don't think it matters in XML files, all parsers that I know of will ignore indents anyway.

czyzby replied:
Reverted, just to be safe.

czyzby commented:
> I find both readable, and if the linter understands it, then using 4 spaces for generated sources should be enough to pass style guides.

The file templates are mostly in strings, and those do not have to follow the style guide (their contents not considered as project's sources other than as multiline strings, and the formatter has no business changing the string values throughout the application).

tommyettinger commented:
Looks good, I think we should be fine with this as it is now. Thanks!
